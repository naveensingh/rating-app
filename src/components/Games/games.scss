@import "../../mixins";

.games {
  @include flex-me;
  display: flex;
  width: 100%;
  height: calc(100vh - 154px);
  overflow: auto;
  flex-wrap: wrap;

  .game {
    flex-basis: 30%;
    height: 160px;
    border-radius: 6px;
    -webkit-box-shadow: 0 8px 32px 0 rgba(3, 22, 51, .2);
    box-shadow: 0 8px 32px 0 rgba(3, 22, 51, .2);
    background-color: #fff;
    display: flex;
    justify-content: flex-start;
    cursor: pointer;
    margin: 16px;
    padding: 20px;
    box-sizing: border-box;

    &--meta {
      padding-left: 10px;
    }

    &--avatar {
      img {
        height: 75px;
        width: 75px;
      }
    }

    &--title {
      font-weight: bold;
      margin-bottom: 5px;
    }

    &--description {
      color: #6681aa;
      font-size: 13px;
    }
  }
}
