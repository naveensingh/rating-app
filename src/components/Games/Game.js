import React from 'react';
import PropTypes from 'prop-types';
import './games.scss';
import { useDispatch } from 'react-redux';
import Rating from '../Rating/Rating';

const Game = ({ game }) => {
  const { uuid, title, description, rating, avatar } = game;
  const dispatch = useDispatch();
  const avatarPath = require(`../../static/images/${avatar}`);
  const setRating = ratingValue => dispatch({
    type: 'SET_RATING',
    payload: {
      uuid,
      title,
      description,
      avatar,
      rating: ratingValue
    }
  });
  return (
    <div className="game" key={uuid}>
      <div className="game--avatar">
        <img src={avatarPath} alt={title} />
      </div>
      <div className="game--meta">
        <div className="game--title">
          {title}
        </div>
        <div className="game--description">
          {description}
        </div>
        <div className="game--rating">
          <Rating data-testid="rating" rating={rating} onClick={setRating} />
        </div>
      </div>
    </div>
  );
};

Game.propTypes = {
  game: PropTypes.shape({
    uuid: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    avatar: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
  }).isRequired
};

export default Game;
