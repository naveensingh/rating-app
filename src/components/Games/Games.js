import React from 'react';
import PropTypes from 'prop-types';
import './games.scss';
import Game from './Game';

const Games = ({ games }) => {
  const sortedGames = Object.keys(games).sort((a, b) => games[b].rating - games[a].rating);
  return (
    <div className="games">
      {sortedGames.map(key => (<Game key={key} game={games[key]} />))}
    </div>
  );
};

Games.propTypes = {
  games: PropTypes.object.isRequired,
};

export default Games;
