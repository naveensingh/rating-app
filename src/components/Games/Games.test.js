import React from 'react';
import Games from './Games';
import {shallow} from "enzyme";
import Game from "./Game";

const games = {
    "someGame": {
        "title": "Game",
        "avatar": "CallOfDuty.jpeg",
        "description": "Lorem ipsum dolor sit amet",
        "rating": 0,
        "uuid": "someGame"
    }
};

describe('Games spec', () => {
    test('should match snapshot for available games', () => {
        const gamesComponent = shallow(<Games games={games}/>);
        expect(gamesComponent).toMatchSnapshot();
    });
    test('should have rendered 1 game', () => {
        const gamesComponent = shallow(<Games games={games}/>);
        const game = gamesComponent.find(Game);
        expect(game).toHaveLength(1);
    });
});
