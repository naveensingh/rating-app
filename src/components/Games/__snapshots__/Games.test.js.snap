// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Games spec should match snapshot for available games 1`] = `
<div
  className="games"
>
  <Game
    game={
      Object {
        "avatar": "CallOfDuty.jpeg",
        "description": "Lorem ipsum dolor sit amet",
        "rating": 0,
        "title": "Game",
        "uuid": "someGame",
      }
    }
    key="someGame"
  />
</div>
`;
